File: desktop/SuperadminLauncher.cs
using System;
using System.Diagnostics;
using System.Windows.Forms;

namespace ShiiEdu.SuperadminLauncher
{
    internal static class Program
    {
        private const string DefaultUrl = "http://127.0.0.1:3100/app/superadmin";

        [STAThread]
        private static void Main()
        {
            var targetUrl = Environment.GetEnvironmentVariable("SHII_EDU_SUPERADMIN_URL");
            if (string.IsNullOrWhiteSpace(targetUrl))
            {
                targetUrl = DefaultUrl;
            }

            try
            {
                Process.Start(new ProcessStartInfo
                {
                    FileName = targetUrl,
                    UseShellExecute = true
                });
            }
            catch (Exception error)
            {
                MessageBox.Show(
                    "Open this URL manually:\n\n" + targetUrl + "\n\n" + error.Message,
                    "Shii-Edu Superadmin",
                    MessageBoxButtons.OK,
                    MessageBoxIcon.Information);
            }
        }
    }
}

File: desktop/superadmin-launcher.js
const { spawn } = require('child_process');

const targetUrl = process.env.SHII_EDU_SUPERADMIN_URL || 'http://127.0.0.1:3100/app/superadmin';

const launch = () => {
  if (process.platform === 'win32') {
    return spawn('cmd', ['/c', 'start', '""', targetUrl], {
      detached: true,
      stdio: 'ignore',
      windowsHide: true,
    });
  }

  if (process.platform === 'darwin') {
    return spawn('open', [targetUrl], {
      detached: true,
      stdio: 'ignore',
    });
  }

  return spawn('xdg-open', [targetUrl], {
    detached: true,
    stdio: 'ignore',
  });
};

try {
  const child = launch();
  child.unref();
  console.log(`Opening Shii-Edu Superadmin: ${targetUrl}`);
} catch (error) {
  console.error(`Open this URL manually: ${targetUrl}`);
  process.exitCode = 1;
}
